contracts/v0.2.0/LiquidityWallet.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "hardhat/console.sol";
+
+contract LiquidityWallet is Ownable {
+    receive() external payable  {}
+
+    function withdraw(address _addrGrtToken, uint256 amount) external onlyOwner {
+        uint256 balance = IERC20(_addrGrtToken).balanceOf(address(this));
+        require(balance >= amount, "Insufficient balance");
+      
+        require(IERC20(_addrGrtToken).transfer(msg.sender, amount), "Transaction Failed");
+    }
+
+    function payOffer(address addrToken, address addrOfferer, uint256 amount) external onlyOwner { 
+        require(IERC20(addrToken).transfer(addrOfferer, amount), "Transaction Failed");
+    }
+
+    function payOfferWithZapier(address addrToken, address addrZapier, address addrOfferer, uint256 amount) external {
```

Comment:
Not so sure if this function is what you mean @tcoratger

Reply:
Added a push in the `liquidity_wallet_review` branch to guide you in this contract with the following functions:
- Constructor to set bot and satellite addresses
- Functions to set bot and satellite (only owner)
- Withdrawal functions for ERC20 and native tokens
- Pay the offer with native or ERC20 (depends on the offer) as the owner or a a designed bot

You can start from there and implement the missing stuffs if needed + make the tests for that

Reply:
@tcoratger thanks ! 
I have already pushed the changes.